metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -2834,17 +2834,37 @@ class MetalsLspService(
     }
   }
 
+  private def clearBazelDir(folder: AbsolutePath): Unit = {
+    try {
+      folder.resolve(Directories.bazelBsp).deleteRecursively()
+      folder.resolve(Directories.bsp).deleteRecursively()
+    } catch {
+      case e: Throwable =>
+        languageClient.showMessage(Messages.ResetWorkspaceFailed)
+        scribe.error(
+          "Error while deleting directories inside .bsp and .bazelbsp",
+          e,
+        )
+    }
+  }
+
   def resetWorkspace(): Future[Unit] =
     for {
       _ <- disconnectOldBuildServer()
-      _ = optProjectRoot match {
+      shouldImport = optProjectRoot match {
         case Some(path) if buildTools.isBloop(path) =>
           bloopServers.shutdownServer()
           clearBloopDir(path)
-        case _ =>
+          false
+        case Some(path) if buildTools.isBazelBsp =>
```

Comment:
Maybe we should clear `.bsp` for all build servers?

Reply:
Done!